src/app/shared/components/carousel/carousel.component.ts

```diff
@@ -1,58 +1,63 @@
-import {
-  Component,
-  ElementRef,
-  OnInit,
-  ViewChild,
-} from '@angular/core';
-import {CarouselItem} from '../../../carouselItem';
-import {carouselItems} from '../../../carouselItems';
-
-
-import {from, Observable } from 'rxjs';
-import { map } from 'rxjs/operators'
+import { Component, ElementRef, ViewChild, Input } from '@angular/core';
+import { CarouselItem } from '../../../carouselItem';
 
 @Component({
   selector: 'app-carousel',
   templateUrl: './carousel.component.html',
   styleUrls: ['./carousel.component.scss'],
 })
-export class CarouselComponent implements OnInit{
+export class CarouselComponent {
+  private activeIndex = 0;
 
-  private selectedSlideIndex = 0;
-  private translateX = 0;
-  carouselItems$!: Observable<CarouselItem>;
+  @Input() items!: CarouselItem[];
 
   @ViewChild('cardsContainer')
   private containerElement!: ElementRef<HTMLElement>;
 
-  ngOnInit() {
-    this.carouselItems$ = from(carouselItems);
-  }
-
   moveLeft(): void {
-    if (this.showsNotFirstSlide) {
-      this.translateX += this.slideWidth;
-      this.containerElement.nativeElement.style.transform = `translateX(${this.translateX}px)`;
-      this.selectedSlideIndex--;
+    if (!this.isFirstSlideActive) {
+      this.showPreviousSlide();
     }
   }
 
   moveRight(): void {
-    if (this.showsNotLastSlide) {
-      this.translateX -= this.slideWidth;
-      this.containerElement.nativeElement.style.transform = `translateX(${this.translateX}px)`;
-      this.selectedSlideIndex++;
+    if (!this.isLastSlideActive) {
+      this.showNextSlide();
     }
   }
 
-  private get slideWidth(): number {
-    return this.containerElement.nativeElement.offsetWidth;
+  private showNextSlide(): void {
+    this.containerElement.nativeElement.style.transform = `translateX(${this.nextTranslateXValue}px)`;
```

Comment:
Consider making this a calculated property, like:

```ts
get currentTranslateXValue() {
  return `transformX(${this.sliderWidth * -this.currentSlideIndex}px)`
}
```

and then using it in the template

```html
<div [style.transform]="currentTranslateXValue"></div>
```